src/modules/deal/GenericDeal.ts

```diff
@@ -0,0 +1,186 @@
+import PokemonType from '../enums/PokemonType';
+import Amount from '../wallet/Amount';
+import BerryType from '../enums/BerryType';
+import Item from '../items/Item';
+import { ItemList } from '../items/ItemList';
+import GameHelper from '../GameHelper';
+import { Currency } from '../GameConstants';
+
+export type GenericTraderShopIdentifier = 'Palaeontologist' | 'EverstoneDealer';
+
+type DealCostProfit = {
+    amount: number,
+    hidePlayerInventory?: boolean,
+};
+
+type GemDealCost = {
+    type: 'Gem',
```

Comment:
Would it make sense to have this `type` be an enum rather than string?

Reply:
Resolved